### ISSUE TITLE
Label masking and loss calculation

### ISSUE THREAD

**Issue by Marcosuff** [keras-team/keras#12850]
Hello,

I've been trying to train a multitask regression on a dataset with sparse labels. So far I used the following loss function to mask missing labels so they don't contribute to loss calculation:

    def custom_mse(y_true, y_pred,mask=-1):
        mask = K.cast(K.not_equal(y_true, mask), K.floatx())
        return K.mean(K.square(mask*(y_pred - y_true)), axis=-1)

It seems to work as intended since the loss decreases every training epoch. However, I've seen other approach consisting of assigning weight = 0 to any sample with missing labels 
Reference: (https://github.com/deepchem/deepchem/issues/1175)

Both makes sense to me but is there a preferred way to do it in Keras?
